functional-programming-spreadsheet/index190.js

```diff
@@ -0,0 +1,27 @@
+const infixToFunction = {
+  "+": (x, y) => x + y,
+  "-": (x, y) => x - y,
+  "*": (x, y) => x * y,
+  "/": (x, y) => x / y
+};
+
+const infixEval = (str, regex) =>
+  str.replace(regex, (_, arg1, fn, arg2) =>
+    infixToFunction[fn](parseFloat(arg1), parseFloat(arg2))
+  );
+
+const highPrecedence = str => {
+  const regex = /([0-9.]+)([*\/])([0-9.]+)/;
+  const str2 = infixEval(str, regex);
+  return str === str2 ? str : highPrecedence(str2);
+};
+
+const spreadsheetFunctions = {
+  "": x => x
+};
+
+/*
+Define a function `applyFn` which takes an argument `str`.
```

Comment:
This will need to be broken down into multiple steps. The first could be defining an empty function, and learners can create a variable for each subsequent step.